app/app.gradle.kts

```diff
@@ -65,6 +65,70 @@ tasks.withType<Test> {
     dependsOn(project("test_plugins").tasks["jar"])
 }
 
+val nativeShadowTasks = NativePlatforms.values().map { platform ->
+    tasks.create<ShadowJar>("shadowJar-${platform.platformName}") {
```

Comment:
Consider using `tasks.register<ShadowJar>("shadowJar-${platform.platformName}")` instead.

Reply:
`register` only lets you configure a default Task, not a ShadowJar task like I need here

Reply:
It's introduced in 4.10, which has some API compatibility breaks with 4.9

Reply:
You can still use `tasks.register("shadowJar-${platform.platformName}", Shadow::class.java)` in 4.9.

Reply:
This doesn't seem to work with the publishing extension

Reply:
Oh, right, it will in Gradle 4.10.

I have a stopgap measure that works for gradle 4.9 but it uses an internal API.
```
/**
 * Stopgap measure to support [TaskProvider] when a [PublishArtifact] is needed.
 * [Issue](https://github.com/gradle/gradle/issues/6246).
 */
fun lazyPublishArtifactHack(provider: TaskProvider<*>): PublishArtifact =
    LazyPublishArtifact(provider)
```

Using it lets you do this:
```kotlin
            nativeShadowTasks.forEach {
                artifact(lazyPublishArtifactHack(it)) {
                    classifier = it.classifier
                }
            }
```

Otherwise, if you don't want to use the hack you can do:
```kotlin
            nativeShadowTasks.forEach {
                // TODO: Fix this for Gradle 5.0
                artifact(it.get()) {
                    classifier = it.classifier
                }
            }
```

Reply:
I'd prefer to fix it when the API becomes available

Reply:
Okay.